graphFunctions.js

```diff
@@ -0,0 +1,296 @@
+// Function to reorganize the data to limit parameter constraints
+// Inputs: data, an Array of sub-arrays (defined below),
+//			limit, Integer value of maximum allowed devices,
+//			setting, Integer value depending on structure of input

```

Comment:
I had it described before but you said to put it in the function where the if statement was. So I moved it and now I've copied it back into the description so now it's described twice. Seems redundant to me.